failure in dcx/test_ucp_tag_match.sync_send_unexp_rndv/3

http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3213/label=hpc-test-node2,worker=0/consoleFull
```
14:52:43 [ RUN      ] dcx/test_ucp_tag_match.sync_send_unexp_rndv/3
14:52:43 /scrap/jenkins/scrap/workspace/hpc-ucx-pr/label/hpc-test-node2/worker/0/contrib/../test/gtest/ucp/test_ucp_tag_match.cc:599: Failure
14:52:43 Value of: my_send_req->completed
14:52:43   Actual: false
14:52:43 Expected: true
14:52:43 /scrap/jenkins/scrap/workspace/hpc-ucx-pr/label/hpc-test-node2/worker/0/contrib/../test/gtest/ucp/test_ucp_tag_match.cc:600: Failure
14:52:43 Value of: my_send_req->status
14:52:43   Actual: -112
14:52:43 Expected: UCS_OK
14:52:43 Which is: 0
14:52:43 /scrap/jenkins/scrap/workspace/hpc-ucx-pr/label/hpc-test-node2/worker/0/contrib/../test/gtest/ucp/test_ucp_tag_match.cc:599: Failure
14:52:43 Value of: my_send_req->completed
14:52:43   Actual: false
14:52:43 Expected: true
14:52:43 /scrap/jenkins/scrap/workspace/hpc-ucx-pr/label/hpc-test-node2/worker/0/contrib/../test/gtest/ucp/test_ucp_tag_match.cc:600: Failure
14:52:43 Value of: my_send_req->status
14:52:43   Actual: 80
14:52:43 Expected: UCS_OK
14:52:43 Which is: 0
14:52:44 [1490442763.539055] [hpc-test-node2:42243:0]          mpool.c:38   UCX  WARN  object 0x327afc0 was not returned to mpool ucp_requests
14:52:44 [1490442763.831309] [hpc-test-node2:42243:0]          mpool.c:38   UCX  WARN  object 0x3335000 was not returned to mpool ucp_requests
14:52:44 [1490442764.229077] [hpc-test-node2:42243:0]         rcache.c:284  UCX  WARN  mlx5_0: destroying inuse region 0x3237a70 [0x34c97a0..0x35e1e40] gt- rw ref 1 lkey 0x211b6f rkey 0x211b6f atomic: lkey 0x2135b5 rkey 0x2135b5
14:52:44 [1490442764.229567] [hpc-test-node2:42243:0]         rcache.c:284  UCX  WARN  mlx5_0: destroying inuse region 0x33bf410 [0x35e1e50..0x36fa4f0] gt- rw ref 1 lkey 0x1a8a10 rkey 0x1a8a10 atomic: lkey 0x1e7fb4 rkey 0x1e7fb4
14:52:44 /scrap/jenkins/scrap/workspace/hpc-ucx-pr/label/hpc-test-node2/worker/0/contrib/../test/gtest/common/test.cc:133: Failure
14:52:44 Failed
14:52:44 Got 4 warnings during the test
14:52:44 
[  FAILED  ] dcx/test_ucp_tag_match.sync_send_unexp_rndv/3, where GetParam() = \dc_mlx5 (1164 ms)
...
15:05:16 [  FAILED  ] 1 test, listed below:
15:05:16 [  FAILED  ] dcx/test_ucp_tag_match.sync_send_unexp_rndv/3, where GetParam() = \dc_mlx5
15:05:16 
15:05:16  1 FAILED TEST
15:05:16 
make: *** [test] Error 1
```

--- a/test/gtest/ucp/test_ucp_tag_match.cc
+++ b/test/gtest/ucp/test_ucp_tag_match.cc
@@ -421,7 +421,7 @@ UCS_TEST_P(test_ucp_tag_match, sync_send_unexp_rndv, "RNDV_THRESH=1048576") {
     EXPECT_EQ(sendbuf, recvbuf);
 
     /* sender - get the ATS and set send request to completed */
-    ucp_worker_progress(sender().worker());
+    wait_for_flag(&my_send_req->completed);
 
     EXPECT_TRUE(my_send_req->completed);
     EXPECT_EQ(UCS_OK, my_send_req->status);

--- a/test/gtest/ucp/ucp_test.cc
+++ b/test/gtest/ucp/ucp_test.cc
@@ -105,17 +105,6 @@ void ucp_test::short_progress_loop(int worker_index) const {
     }
 }
 
-void ucp_test::wait_for_flag(volatile size_t *flag, double timeout)
-{
-    ucs_time_t loop_end_limit;
-
-    loop_end_limit = ucs_get_time() + ucs_time_from_sec(timeout);
-
-    while ((ucs_get_time() < loop_end_limit) && (!(*flag))) {
-        short_progress_loop();
-    }
-}
-
 void ucp_test::disconnect(const entity& entity) {
     for (int i = 0; i < entity.get_num_workers(); i++) {
         entity.flush_worker(i);

--- a/test/gtest/ucp/ucp_test.h
+++ b/test/gtest/ucp/ucp_test.h
@@ -125,11 +125,18 @@ protected:
 
     void progress(int worker_index = 0) const;
     void short_progress_loop(int worker_index = 0) const;
-    void wait_for_flag(volatile size_t *flag, double timeout = 10.0);
     void disconnect(const entity& entity);
     void wait(void *req, int worker_index = 0);
     static void disable_errors();
     static void restore_errors();
+
+    template <typename T>
+    void wait_for_flag(volatile T *flag, double timeout = 10.0) {
+        ucs_time_t loop_end_limit = ucs_get_time() + ucs_time_from_sec(timeout);
+        while ((ucs_get_time() < loop_end_limit) && (!(*flag))) {
+            short_progress_loop();
+        }
+    }
 
 private:
     static void set_ucp_config(ucp_config_t *config,

